src/Tasks/Microsoft.NET.Build.Tasks/ResolvePublishAssemblies.cs

```diff
@@ -45,12 +47,20 @@ protected override void ExecuteCore()
             IEnumerable<string> privateAssetsPackageIds = PackageReferenceConverter.GetPackageIds(PrivateAssetsPackageReferences);
             IPackageResolver packageResolver = NuGetPackageResolver.CreateResolver(lockFile, ProjectPath);
 
+            LockFile FilterLockFile = null;
+            if (!string.IsNullOrEmpty(FilterProjectAssetsFile))
+            {
+                FilterLockFile = new LockFileCache(BuildEngine4).GetLockFile(FilterProjectAssetsFile);
```

Comment:
You are right that we are using 2 assets files.  But what I'm saying is there is no point in creating two instances of the `LockFileCache` object.  The same cache object can load multiple asset files.